File: #80_Remove_Duplicates_from_Sorted_Array_II/Solution.java
public class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums == null || nums.length == 0) return 0;
        int cp = 1, cur = 1, count = 0;
        while (cur < nums.length) {
        	if (nums[cur] == nums[cur - 1]) count++;
        	else count = 0;
        	if (count < 2) {
        		nums[cp++] = nums[cur];
        	}
        	cur++;
        }
        return cp;
    }

    public static void main(String[] args) {
    	Solution sol = new Solution();
    	int[] t = {1, 1, 1, 1, 1, 1};
    	System.out.println(sol.removeDuplicates(t));
    }
}

File: #80_Remove_Duplicates_from_Sorted_Array_II/Solution.py
'''
dup can be set to 0 only when nums[j] is not equal to nums[i - 1]
'''

class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def removeDuplicates(self, nums):
        if not nums:
        	return 0
        i, j, dup = 0, 0, 0
        while j < len(nums) and i <= j:
        	if i > 0 and nums[i - 1] == nums[j]:
        		dup += 1
        	else:
        		dup = 0
        	if dup >= 2:
        		j += 1
        	else:
        		nums[i] = nums[j]
        		i += 1
        		j += 1
        return i

sol = Solution()
print sol.removeDuplicates([1, 1, 1, 1, 2, 2, 3])